apps/cowswap-frontend/src/modules/swap/containers/SwapConfirmModal/index.tsx

```diff
@@ -54,13 +53,19 @@ export function SwapConfirmModal(props: SwapConfirmModalProps) {
   const submittedContent = useOrderSubmittedContent(chainId)
   const labelsAndTooltips = useLabelsAndTooltips()
 
+  const primaryFormValidation = useGetTradeFormValidation()
+
+  // Selling ETH is allowed in Swap
+  const isPrimaryValidationPassed =
+    !primaryFormValidation || primaryFormValidation === TradeFormValidation.SellNativeToken
+
   const buttonText = useMemo(() => {
-    if (!hasEnoughWrappedBalanceForSwap) {
+    if (!isPrimaryValidationPassed) {
```

Comment:
Should be `if (primaryFormValidation === TradeFormValidation.SellNativeToken ? false : hasEnoughWrappedBalanceForSwap)`, no?
Otherwise we will display `Insufficient Balance` always when selling ETH

Reply:
Or just:

```
const current = inputCurrencyInfo?.amount?.currency

if (current && !getIsNativeToken(current) && !hasEnoughWrappedBalanceForSwap) {...
```